### PR TITLE
scrounge-ntfs: init at 0.9

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Description of changes
Note: the file source URL is only HTTP.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by D3vil0p3r** [comment]
Result of `nixpkgs-review pr 288039` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scrounge-ntfs</li>
  </ul>
</details>

**Review by AkechiShiro — APPROVED** [review]
- Did a nixpkgs-review on this PR, package was built successfully.
- Tested basic functionality of binary.
LGTM

File: pkgs/by-name/sc/scrounge-ntfs/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  nativeBuildInputs = [ autoreconfHook ];
+
+  meta = with lib; {
+    description = "Data recovery program for NTFS file systems";
+    mainProgram = "scrounge-ntfs";
+    homepage = "http://thewalter.net/stef/software/scrounge/";
+    maintainers = with maintainers; [ d3vil0p3r ];
+    platforms = platforms.unix;
+    license = licenses.unfree;
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/sc/scrounge-ntfs/package.nix:24]: home page states:
```
Open Source BSD type license
```
http://thewalter.net/git/cgit.cgi/scrounge-ntfs/tree/COPYING
so the license seems to be free, though offhand i am not sure what the proper mechanism is to specify one-of licenses.
https://nixos.org/manual/nixpkgs/unstable/#sec-meta-license

Comment (D3vil0p3r) [reply pkgs/by-name/sc/scrounge-ntfs/package.nix:24]: Mmh actually also the [BlackArch PKGBUILD](https://github.com/BlackArch/blackarch/blob/master/packages/scrounge-ntfs/PKGBUILD) sets the license as BSD.

Comment (D3vil0p3r) [reply pkgs/by-name/sc/scrounge-ntfs/package.nix]: @a-n-n-a-l-e-e used `bsd0`.

Comment (a-n-n-a-l-e-e) [reply pkgs/by-name/sc/scrounge-ntfs/package.nix]: `bsd3` https://opensource.org/license/bsd-3-clause/ matches better

Comment (D3vil0p3r) [reply pkgs/by-name/sc/scrounge-ntfs/package.nix]: Fixed

File: pkgs/by-name/sc/scrounge-ntfs/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, stdenv
+, fetchurl
+, autoreconfHook
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/sc/scrounge-ntfs/package.nix:4]: not needed

File: pkgs/by-name/sc/scrounge-ntfs/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+stdenv.mkDerivation (finalAttrs: {
+  pname = "scrounge-ntfs";
+  version = "0.9";
+
+  src = fetchurl {
+    url = "http://thewalter.net/stef/software/scrounge/scrounge-ntfs-${finalAttrs.version}.tar.gz";
+    hash = "sha256-HYrMIMTRPmgAac/vaZ1jaUFchyAl5B0quxgHH0DHJ84=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/sc/scrounge-ntfs/package.nix:16]: there is already a configure script so the build will work without this. is there a reason you're adding this hook?

**Review by ghost — COMMENTED** [review]
can remove auotreconfHook as there is a configure script.

darwin doesn't build but if you patch with the diff below it will. [editing] can't paste diff in a text box .. attached in comment below.

**Comment by a-n-n-a-l-e-e** [comment]
here is the diff -- copy / pasting didn't get the ^M chars needed to apply it.
[darwin.diff.gz](https://github.com/NixOS/nixpkgs/files/14318737/darwin.diff.gz)

**Comment by D3vil0p3r** [comment]
> here is the to diff so darwin builds -- copy / pasting didn't get the ^M chars needed to apply it. [darwin.diff.gz](https://github.com/NixOS/nixpkgs/files/14318737/darwin.diff.gz)

Should I add it as patch?

**Comment by a-n-n-a-l-e-e** [comment]
> > here is the to diff so darwin builds -- copy / pasting didn't get the ^M chars needed to apply it. [darwin.diff.gz](https://github.com/NixOS/nixpkgs/files/14318737/darwin.diff.gz)
> 
> Should I add it as patch?

i think so -- makes the darwin build work. it is pretty minor, just fixes an ifdef and a couple of includes.

**Comment by D3vil0p3r** [comment]
> > > here is the to diff so darwin builds -- copy / pasting didn't get the ^M chars needed to apply it. [darwin.diff.gz](https://github.com/NixOS/nixpkgs/files/14318737/darwin.diff.gz)
> > 
> > 
> > Should I add it as patch?
> 
> i think so -- makes the darwin build work. it is pretty minor, just fixes an ifdef and a couple of includes.

Patch for darwin added.

**Review by ghost — APPROVED** [review]
Result of `nixpkgs-review pr 288039` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scrounge-ntfs</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 288039` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scrounge-ntfs</li>
  </ul>
</details>

**Pull request merged**